services/httpd/service.go

```diff
@@ -192,6 +197,16 @@ func (s *Service) Open() error {
 func (s *Service) Close() error {
 	s.Handler.Close()
 
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	s.httpServerMutex.Lock()
+	// shut down all running http servers
+	for _, server := range s.httpServer {
+		server.Shutdown(ctx)
```

Comment:
This may be overkill for the scenario, but I've seen timed cleanups/shutdowns in separate threads per instance, so one blocking doesn't time out all the others being closed.  That does require a counting semaphore or something similar, rather than a simple mutex.   Or, you may want to scale your timeout on the number of Shutdown() calls being made.

Reply:
this is good feedback.  thank you!  will make those changes.